
PACKAGE = "imu_clean"

import dynamic_reconfigure.parameter_generator_catkin as pg

gen = pg.ParameterGenerator()
# 滑动窗口大小（1-100帧）
gen.add("window_size", pg.IntParam, 0, "Sliding window size for IMU data cleaning", 10, 1, 100)
# 标准差阈值（0.01-1.0）
gen.add("std_threshold", pg.DoubleParam, 0, "Standard deviation threshold for outlier detection", 0.1, 0.01, 1.0)
# 是否将陀螺仪单位转为deg/s（默认rad/s）
gen.add("gyro_to_deg", pg.BoolParam, 0, "Convert gyro from rad/s to deg/s", False)

exit(gen.generate(PACKAGE, "imu_clean", "ImuClean"))